compiler/src/main/java/org/robovm/compiler/AppCompiler.java

```diff
@@ -241,10 +244,39 @@ private boolean compile(Executor executor, ClassCompilerListener listener,
                     compileQueue.add(depClazz);
                 }
             }
+            addMetaInfImplementations(config.getClazzes(), clazz, compiled, compileQueue);
         }
         return result;
     }
 
+    static void addMetaInfImplementations(Clazzes clazzes, Clazz clazz, Set<Clazz> compiled, Set<Clazz> compileQueue) throws IOException {
+        String metaInfName = "META-INF/services/" + clazz.getClassName();
+        IOException throwLater = null;
+        for (InputStream is : clazzes.loadResources(metaInfName)) {
+            try (BufferedReader r = new BufferedReader(new InputStreamReader(is))) {
```

Comment:
Perhaps use an explicit charset name here? UTF-8? Unlikely to become a problem unless the user has some weird default charset like UTF-16.